doc/source/conf.py

```diff
@@ -253,26 +253,33 @@ def setup(app):
 # Additional stuff for the LaTeX preamble.
 latex_elements['preamble'] = r'''
 % In the parameters section, place a newline after the Parameters
-% header
-\usepackage{xcolor}
+% header.  This is default with Sphinx 5.0.0+, so no need for
+% the old hack then.
+% Unfortunately sphinx.sty 5.0.0 did not bump its version date
+% hence some complications here:  we first check if Sphinx is
+% at 4.0.0+ so sphinxpackagefootnote.sty exists then check
+% the latter date which was bumped to 2022/02/12 at Sphinx 5.0.0.
+\makeatletter
+\if1\@ifpackagelater{sphinx}{2021/01/27}% later = at least
+    {\@ifpackagelater{sphinxpackagefootnote}{2022/02/12}01}%
+    {1}%
+% this until next \fi gets executed if and only if Sphinx < 5.0.0
 \usepackage{expdlist}
 \let\latexdescription=\description
 \def\description{\latexdescription{}{} \breaklabel}
 % but expdlist old LaTeX package requires fixes:
 % 1) remove extra space
 \usepackage{etoolbox}
-\makeatletter
 \patchcmd\@item{{\@breaklabel} }{{\@breaklabel}}{}{}
-\makeatother
 % 2) fix bug in expdlist's way of breaking the line after long item label
-\makeatletter
 \def\breaklabel{%
     \def\@breaklabel{%
         \leavevmode\par
         % now a hack because Sphinx inserts \leavevmode after term node
         \def\leavevmode{\def\leavevmode{\unhbox\voidb@x}}%
     }%
 }
+\fi
```

Comment:
```suggestion
\fi  % sphinx < 5.0.0
```
Could make it like we do in headers often.

Our `doc_requirements.txt` says `sphinx>=4.5` currently, so it seems OK to even remove the 4.0.0+ check?
(honestly, I would be fine with just refusing to do the whole pdf build on old sphinx.)

Reply:
I have applied the suggested change.

> Our doc_requirements.txt says sphinx>=4.5 currently, so it seems OK to even remove the 4.0.0+ check?
(honestly, I would be fine with just refusing to do the whole pdf build on old sphinx.)

ah sorry I missed that.  So indeed the test can be simplified with only one `\@ifpackagelater` and then the LaTeX can look simpler with no `\if...\fi`.  I am sorry we did not at Sphinx 5.0.0 bump the sphinx.sty date stamp, this is a recurring problem.

Should I simplify the test then?

Reply:
If it looks much nicer, I would lean to yes.  But whichever way you like, honestly.  I am not sure what happens if you try to build our docs on an older sphinx, but anyone using a "blessed" workflow shouldn't.

Reply:
Regarding `numpy 1.24.1` which is the one installed in my `Python 3.10.9`  virtual environment (together with `breathe` and `scipy`  and other dependencies), I observe the following

- Sphinx `4.0.0` and `4.1.0`: ``ImportError: cannot import name 'Union' from 'types' (/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/types.py)``
- Sphinx `4.2.0`: ``This project needs at least Sphinx v4.3 and therefore cannot be built with this version.``
- Sphinx `4.3.0`:  `make latex` build appears to complete with no error and I can then issue `make -C build/latex all-pdf` with success.

All versions of Sphinx `>=4.3.0` which I tested appear to work for PDF build.  I think they were `4.5.0`, Sphinx `5.0.0`, Sphinx `5.1.0` and Sphinx `6.0.0.` and finally `6.1.0` (released today).

I will push a commit to simplify the Boolean logic in checking the Sphinx version as I can assume it is already at least 4.0.0.

Reply:
I am now happy...